spec/oauth2/client_spec.rb

```diff
@@ -544,6 +548,27 @@ def self.contains_token?(hash)
           expect { client.get_token({}, {}, nil, access_token_class: CustomAccessToken) }.to raise_error(OAuth2::Error)
         end
       end
+
+      context 'when status code is 200' do
+        let(:status_code) { 200 }
+
+        context 'when the request body is not nil' do
+          let(:body) { JSON.dump('access_token' => 'the-token') }
+
+          it 'raises an error' do
+            expect { client.get_token({}) }.to raise_error(OAuth2::Error)
```

Comment:
Hmm, I think I meant to keep this test case as a 500 error? This does not seem right.